### DISCUSSION TITLE
Structure.get_subgraphs_as_molecules() breaks molecules

### DISCUSSION THREAD

**Discussion by OphirKa** [materialsproject/pymatgen#2628] [Q&A]
I am using Structure.get_subgraphs_as_molecules() to have the molecule in one-molecular structure (So this function suppose to return only one molecule),but for some cif files I get some seperated molecules that are parts from the original molecules.
 `
struct = CifParser(os.path.join("data", refcode + ".cif"),occupancy_tolerance=100).get_structures()[0]
sg = StructureGraph.with_local_env_strategy(struct, JmolNN())

    # Extract individual molecules, which are isolated subgraphs
    # (i.e. groups of atoms not connected to each other)
    molecules = sg.get_subgraphs_as_molecules()
`
One problematic cif file is attached: 
[crystal.txt](https://github.com/materialsproject/pymatgen/files/9389118/crystal.txt)
.

Thanks

**Comment by janosh** [comment]
Hav you tried tweaking the `tol` and `el_radius_updates` kwargs to `JmolNN`?

<details>
<summary>Doc string</summary>

```py
class JmolNN(NearNeighbors):
    """
    Determine near-neighbor sites and coordination number using an emulation
    of Jmol's default autoBond() algorithm. This version of the algorithm
    does not take into account any information regarding known charge
    states.
    """

    def __init__(
        self, tol: float = 0.45, min_bond_distance: float = 0.4, el_radius_updates: dict[SpeciesLike, float] = None
    ):
        """
        Args:
            tol (float): tolerance parameter for bond determination
                Defaults to 0.56.
            min_bond_distance (float): minimum bond distance for consideration
                Defaults to 0.4.
            el_radius_updates: (dict) symbol->float to override default atomic
                radii table values
        """
```
</details>

**Reply by OphirKa** [reply]
Thank you for your answer. How can I know what parameters I should use? I tried a wide range (for tol - .5-1.1) and didn't found any value that fit for all of the crystals. I also tried to check the problematic molecules bonds length and compare to the bonds according bonds_jmol_ob.yaml and did not find any irregularity.
